docs/src/components/ComponentDoc/ComponentSourceManager/ComponentSourceManager.ts

```diff
@@ -111,7 +99,12 @@ export default class ComponentSourceManager extends React.Component<
   }
 
   handleCodeFormat = (): void => {
-    this.setState(prevState => ({ currentCode: prevState.formattedCode }))
+    const { currentCode, currentCodeLanguage } = this.state
```

Comment:
This means that whenever the users click this button, we will prettify it, we just won't show them when it should be prettified right?

Reply:
Exactly

Reply:
Not super-friendly, but it's better than prettify on each render to show the status of a single button...

Reply:
Sure, totally agree. I was just checking if I understand this correctly..

Reply:
Prettify on each render is bad, agree.
But to check if code **changed since it was formatted** would be cheap and way better than `wasCodeChanged` which checks if current code differs from the original code.

Reply:
Reworked this stuff as you proposed 👍